release/pipline/dags/istio_common_dag.py

```diff
@@ -0,0 +1,251 @@
+    GCS_PATH = conf.get('GCS_PATH') or default_conf['GCS_PATH']
+    GCS_DEST = conf.get('GCR_DEST') or default_conf['GCR_DEST']
+
+    SVC_ACCT = conf.get('SVC_ACCT') or default_conf['SVC_ACCT']
+    GITHUB_ORG = conf.get('GITHUB_ORG') or default_conf['GITHUB_ORG']
+    GITHUB_REPO = conf.get('GITHUB_REPO') or default_conf['GITHUB_REPO']
+    GCS_GITHUB_PATH = conf.get(
+        'GCS_GITHUB_PATH') or default_conf['GCS_GITHUB_PATH']
+
+    TOKEN_FILE = conf.get('TOKEN_FILE') or default_conf['TOKEN_FILE']
+    GCR_DEST = conf.get('GCR_DEST') or default_conf['GCR_DEST']
+    GCS_DEST = conf.get('GCS_DEST') or default_conf['GCS_DEST']
+    DOCKER_HUB = conf.get('DOCKER_HUB') or default_conf['DOCKER_HUB']
+    BUILD_GCS_BUCKET = conf.get(
+        'BUILD_GCS_BUCKET') or default_conf['BUILD_GCS_BUCKET']
+    return {
+        'PROJECT_ID': PROJECT_ID,
+        'MFEST_URL': MFEST_URL,
+        'MFEST_FILE': MFEST_FILE,
+        'MFEST_COMMIT': MFEST_COMMIT,
+        'execution_date': date,
+        'VERSION': VERSION,
+        'VERSION_TUPLE': (major_version, minor_version, patch, date),
+        'GCR_BUCKET': GCR_BUCKET,
+        'GCS_BUCKET': GCS_BUCKET,
+        'GCS_PATH': GCS_PATH,
+        'BUILD_GCS_BUCKET': BUILD_GCS_BUCKET,
+        'BUILD_GCS_PATH': '{}/{}'.format(BUILD_GCS_BUCKET, GCS_PATH),
+        'GCS_SOURCE': '{}/{}'.format(BUILD_GCS_BUCKET, GCS_PATH),
+        'GCR_DEST': GCR_DEST,
+        'GCS_DEST': GCS_DEST,
+        'SVC_ACCT': SVC_ACCT,
+        'GITHUB_ORG': GITHUB_ORG,
+        'GITHUB_REPO': GITHUB_REPO,
+        'GCS_GITHUB_PATH': GCS_GITHUB_PATH,
+        'TOKEN_FILE': TOKEN_FILE,
+        'Docker_HUB': DOCKER_HUB,
+    }
+
+
+
+  def GetSettingTemplate(setting):
+    return "{{ task_instance.xcom_pull(task_ids='generate_workflow_args').%s }}" % (
+        setting)
+
+  generate_flow_args = PythonOperator(
+      task_id='generate_workflow_args',
+      python_callable=GenerateTestArgs,
+      provide_context=True,
+      dag=common_dag,
+  )
+
+  def GetSettingPython(ti, setting):
+    return it.xcom_pull(task_id=generate_flow_args.task_id)[setting]
+
+  get_git_commit_cmd = """
+    {% set settings = task_instance.xcom_pull(task_ids='generate_workflow_args') %}
+    git config --global user.name "TestRunnerBot"
+    git config --global user.email "testrunner@example.com"
+    git clone {{ settings.MFEST_URL }} green-builds || exit 2
+    cd green-builds
+    git checkout {{ settings.MFEST_COMMIT }} || exit 3
+    git rev-parse HEAD
+    """
+
+  get_git_commit = BashOperator(
+      task_id='get_git_commit',
+      bash_command=get_git_commit_cmd,
+      xcom_push=True,
+      dag=common_dag)
+
+  build_template = """
+    chmod +x /home/airflow/gcs/data/release/*
+    {% set settings = task_instance.xcom_pull(task_ids='generate_workflow_args') %}
+    {% set m_commit = task_instance.xcom_pull(task_ids='get_git_commit') %}
+    /home/airflow/gcs/data/release/start_gcb_build.sh -w -p {{ settings.PROJECT_ID \
```

Comment:
It's be more ideal and desirable to be fetching the scripts from git (akin to what's done above just to query the latest commit for a branch) rather than storing a static set of files.

Reply:
Agreed, another alternative is to first fetch all latest scripts from git to gcs bucket.

Reply:
I'm not sure about that. I would like to use the commit appropriate version of the files using the method I use above. However my understanding is that your cloud builder scripts always checkout from head and not from the hash I provide. Changing this would be the work of a future cl.

Putting the scripts in a gcs bucket is uglily because I would have to redownload them every steep that uses them. That said I could do it.

Reply:
The repo tool is used by the build to populate the docker image, and the scripts run inside the image are based on the specified manifest.  However, the start_gcb_[build|tag|publish].sh scripts don't use repo tool to perform updates.  In the first version the start_gcb_build.sh did use the repo tool to grab the template, but by the second version (where the tag and publish scripts were added) I felt like these scripts were not sufficiently thin and that if the whole repo is grabbed then it'd be more consistent & reliable to just grab & run both the script and the template from the repo indicated by the manifest.  That's why the example I provided for the second version had its own calls to the repo tool:

BLD_TEMP_DIR="$(mktemp -d /tmp/build.temprepo.XXXX)"
ISTIO_REPO_PATH="go/src/istio.io/istio"
curl -s -L -o "${BLD_TEMP_DIR}/repo" https://storage.googleapis.com/git-repo-downloads/repo
chmod u+x "${BLD_TEMP_DIR}/repo"

pushd ${BLD_TEMP_DIR}
./repo init -u "${SCRIPT_MFEST_URL}" -m "${SCRIPT_MFEST_FILE}" -b "${SCRIPT_MFEST_COMMIT}"
./repo sync "${ISTIO_REPO_PATH}"

$ISTIO_REPO_PATH/release/start_gcb_build.sh ...

Reply:
Cool, I'll open an issue to do this!